src/main/java/carpet/script/value/Value.java

```diff
@@ -79,6 +81,14 @@ public String getPrettyString()
 
     public Value add(Value o) {
         String lstr = this.getString();
+        if (o instanceof FormattedTextValue)
```

Comment:
That's the only part I am little squinting about, but I guess its fine. '+' on non-numerical values is rare and doesn't need to be super efficient, so adding special cases is I guess fine.